src/background/listener/bee-api.listener.ts

```diff
@@ -16,6 +20,45 @@ export class BeeApiListener {
     return this._beeApiUrl
   }
 
+  /**
+   * Handles postage batch id header replacement with global batch id
+   */
+  private globalPostageStampHeaderListener = (
+    details: chrome.webRequest.WebRequestHeadersDetails,
+  ): void | chrome.webRequest.BlockingResponse => {
+    if (!this._globalPostageBatchEnabled || !details.requestHeaders) return
+
+    const postageBatchIdHeaderIndex = details.requestHeaders.findIndex(
+      header => header.name === 'swarm-postage-batch-id',
+    )
```

Comment:
Can we use `.find` here?

Then instead of `if (postageBatchIdHeaderIndex === -1) return` you can use `if (!postageBatchIdHeader)`

Then later on instead of `details.requestHeaders[postageBatchIdHeaderIndex].value`, just use `postageBatchIdHeader.value`